actionmailer/lib/rails/generators/mailer/mailer_generator.rb

```diff
@@ -4,16 +4,22 @@ class MailerGenerator < NamedBase
       source_root File.expand_path("../templates", __FILE__)
 
       argument :actions, type: :array, default: [], banner: "method method"
-      check_class_collision
+
+      check_class_collision suffix: "Mailer"
 
       def create_mailer_file
-        template "mailer.rb", File.join('app/mailers', class_path, "#{file_name}.rb")
+        template "mailer.rb", File.join('app/mailers', class_path, "#{file_name}_mailer.rb")
         if self.behavior == :invoke
           template "application_mailer.rb", 'app/mailers/application_mailer.rb'
         end
       end
 
       hook_for :template_engine, :test_framework
+
+      protected
+        def file_name
+          @_file_name ||= super.gsub(/\_mailer/i, '')
```

Comment:
I've tried your suggestion @JuanitoFatas , but looks like using `sub` this way doesn't account for removing duplicates. See the failure on _MailerGeneratorTest#test_mailer_suffix_is_not_duplicated_ at https://travis-ci.org/rails/rails/jobs/46081505